File: web-frontend/types/bundle.ts
import { BundleStatus } from '../config/status';
import { User } from './user';

export interface Bundle {
  id: string;
  supplierId: string;
  title: string;
  description: string;
  price: number;
  status: BundleStatus;
  images: string[];
  category: string;
  condition: string;
  weight: number;
  dimensions: {
    length: number;
    width: number;
    height: number;
  };
  createdAt: Date;
  updatedAt: Date;
  supplier?: User;
}

export interface BundleItem {
  id: string;
  bundleId: string;
  name: string;
  description: string;
  quantity: number;
  condition: string;
  estimatedValue: number;
  images: string[];
}

export interface CreateBundleInput {
  title: string;
  description: string;
  price: number;
  images: string[];
  category: string;
  condition: string;
  weight: number;
  dimensions: {
    length: number;
    width: number;
    height: number;
  };
  items: Omit<BundleItem, 'id' | 'bundleId'>[];
}

export interface UpdateBundleInput extends Partial<CreateBundleInput> {
  status?: BundleStatus;
} 

File: mobile_frontend/lib/features/reseller/marketplace/domain/entities/bundle.dart
class Bundle {
  final String id;
  final String title;
  final String description;
  final String sampleImage;
  final int quantity;
  final String grade;
  final String sortingLevel;
  final Map<String, int> estimatedBreakdown;
  final String type;
  final double price;
  final String status;
  final DateTime createdAt;
  final int declaredRating;
  final int remainingItemCount;
  final String supplierId;
  final double supplierRating;
  final String sizeRange;
  final List<String> clothingTypes;

  Bundle({
    required this.id,
    required this.title,
    required this.description,
    required this.sampleImage,
    required this.quantity,
    required this.grade,
    required this.sortingLevel,
    required this.estimatedBreakdown,
    required this.type,
    required this.price,
    required this.status,
    required this.createdAt,
    required this.declaredRating,
    required this.remainingItemCount,
    required this.supplierId,
    required this.supplierRating,
    required this.sizeRange,
    required this.clothingTypes,
  });
}